app/src/middleware/authentication.js

```diff
@@ -116,7 +116,15 @@ const currentUser = async (req, res, next) => {
           throw new Error('OIDC environment variable KC_PUBLICKEY or keycloak.publicKey must be defined');
         }
 
-        if (isValid) {
+        // ensure token contains supported idp claim
+        const decodedToken = jwt.decode(bearerToken);
```

Comment:
I think you can reuse `isValid` from further up (albeit with a possible rename for readability) - it already returns the decoded token as long as the JWT is valid:

https://github.com/bcgov/common-object-management-service/blob/5224413f96dafd41a497a0f78679cb56b61d9d22/app/src/middleware/authentication.js#L112-L114